chore: update readme with actionable advice (#143)
es-tooling/ecosystem-cleanup: README.md

Reviewer: Hey I really like that you're documenting processes like this! (thanks @43081j for pointing me to this PR) ❤️ 

A point of feedback as a maintainer who's received quite a few PRs from e18e/ecosystem-cleanup initiatives: please tell participants to _always follow repository contributing guidelines_. I've seen quite a few PRs that don't address accepted issues, have failing CI, don't use the right conventional commit PR titles, etc.

There are two reasons why I think this is important:

* This initiative is more than just rote dependency swap-outs. It's also about convincing the ecosystem at large that this is important enough to spend time on. If people affiliated with an initiative consistently trample over the PRs they're trying to "help" then it looks like it's just mindless enthusiasm - not something thought deeply on by people who understand open source and maintenance. That devalues the initiative and biases us maintainers against it.
* Those processes are there for a reason! Oftentimes there are multiple possible replacements for a package. Projects often need time to discuss and figure out their preferred replacement strategy. Sending a PR early can be wasted work.

To be clear, I'm in favor of cleaning up the ecosystem. It's valuable work and I'm happy to see so much progress on it. I hope this comes across as constructive feedback and trying to help, not external criticism. 🙂

Author: Thanks for the feedback! I tried to incorporate it in the last commit (even stole a few of your words - hope you dont mind :D). Can you check if its better now?

Reviewer: Lovely! I don't mind at all haha, thank you!

Our goal is to provide a cleaner, faster and simpler dependency tree in the JS
ecosystem for those who are concerned with it.

> ## How to help

Reviewer 2: This section kind of duplicates the next one. The only one here that isn't below is "needs alternative". I wonder if we can remove this section and potentially add that one below

Author: While you are right, that it is a bit repetitive, I am in favor of a bit more hand-holding for the start. So people are really sure, they know what they need to do. I would like to leave it but ofc I will change it if nobody likes my opinion on that topic :D

All issues on our issue tracker are labeled and a lot of them contain actionable items you can start to work on right away.
Here are some examples:

<img src="https://img.shields.io/badge/needs%20first%20contact-b60205?style=flat" alt="needs first contact" style="vertical-align: middle"/>

> The maintainer of the repo was not contacted yet. Create an issue and ask, if creating a PR is welcome. You can also reach out to the maintainer via other channels but **make sure that contributions are welcome**. We want the maintainers on board and convince them that this is important enough to spend time on.

Reviewer: ```suggestion
The maintainer of the repo was not contacted yet. Create an issue and ask per the repo's contributing guidelines.

If an issue was filed more than a couple months ago and hasn't received activity, you can also reach out to the maintainer via other channels listed on the repo but **make sure that contributions are welcome**. We want the maintainers on board and to convince them that this is important enough to spend time on.
```

<img src="https://img.shields.io/badge/accepts%20prs-DBE1EF?style=flat" alt="accepts prs" style="vertical-align: middle"/>

The maintainer is welcoming PRs. Feel free to create PRs for all the issues you find. **Make sure to follow contribution guidelines!**

<img src="https://img.shields.io/badge/needs%20alternative-F32096?style=flat" alt="needs alternative" style="vertical-align: middle"/>

> The package in question needs a replacement that we can bring forward. Feel free to create one. Maitenance burden is on you!

Reviewer: ```suggestion
The package in question needs a replacement that we can bring forward. Feel free to create one, but beware: its maintenance burden is on you!
```

ecosystem for those who are concerned with it.

## How to help

All issues on our issue tracker are labeled and a lot of them contain actionable items you can start to work on right away.
Here are some examples:

<img src="https://img.shields.io/badge/needs%20first%20contact-b60205?style=flat" alt="needs first contact" style="vertical-align: middle"/>

The maintainer of the repo was not contacted yet. Create an issue and ask per the repo's contributing guidelines.

If an issue was filed more than a couple months ago and hasn't received activity, you can also reach out to the maintainer via other channels listed on the repo but **make sure that contributions are welcome**. We want the maintainers on board and to convince them that this is important enough to spend time on.

<img src="https://img.shields.io/badge/accepts%20prs-DBE1EF?style=flat" alt="accepts prs" style="vertical-align: middle"/>

The maintainer is welcoming PRs. Feel free to create PRs for all the issues you find. **Make sure to follow contribution guidelines!**

<img src="https://img.shields.io/badge/needs%20alternative-F32096?style=flat" alt="needs alternative" style="vertical-align: middle"/>

> The package in question needs a replacement that we can bring forward. Feel free to create one, but beware: its maintenance burden is on you!

Reviewer 3: I'll bring this one up in discord but the tldr is I think we should say something like: 

> You should look for a replacement package and, if an appropriate one doesn't exist, consider making a new one. Though keep in mind new packages should be well tested and the community made aware of them before any migrations

Make sure to test the new package thoroughly and promote it within the community to gain visibility and trust. Avoid migrating any repositories to your new package without first double-checking with the community.

## Where do I start

- If you'd like to reach out, look for issues with the <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22needs+first+contact%22"><img src="https://img.shields.io/badge/needs%20first%20contact-b60205?style=flat" alt="needs first contact" style="vertical-align: middle"/></a> label and create an issue (or use some other communication channel). Paste the issue you created as answer to tagged issue here
- If you want to create PRs, look for the <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22accepts+prs%22"><img src="https://img.shields.io/badge/accepts%20prs-DBE1EF?style=flat" alt="accepts prs" style="vertical-align: middle"/></a> label. Paste the PR link in the tagged issue.

> - if you create an issue or pr for an <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22umbrella+issue%22"><img src="https://img.shields.io/badge/umbrella%20issue-30EF5D?style=flat" alt="umbrella issue" style="vertical-align: middle"/></a> issue, update the umbrella issue as well

Reviewer 2: The other bullets start with capital letters. I don't have a preference in either direction, but we should probably be consistent
```suggestion
- If you create an issue or pr for an <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22umbrella+issue%22"><img src="https://img.shields.io/badge/umbrella%20issue-30EF5D?style=flat" alt="umbrella issue" style="vertical-align: middle"/></a> issue, update the umbrella issue as well
```

> - This is no must, but we usually try to follow the following process:

Reviewer 2: ```suggestion
- This is not required, but we often try to follow the following process:
```

Our goal is to provide a cleaner, faster and simpler dependency tree in the JS
ecosystem for those who are concerned with it.

## How to help

All issues on our issue tracker are labeled and a lot of them contain actionable items you can start to work on right away.

> Hence, an explanation of the labels is in order:

Reviewer 2: I wouldn't add an explanation for all the labels as it's likely to get out-of-sync as we add, remove, update labels. The labels already have descriptions that are shown on GitHub when the label is hovered.

I think it'd be okay to call out one or two labels here like "accepts prs" that might help folks get started more quickly

<img src="https://img.shields.io/badge/has%20pr-6E4234?style=flat" alt="has pr" style="vertical-align: middle"/>

A PR has been created and waits to be merged. Nothing to do here for now.

<img src="https://img.shields.io/badge/pr%20needs%20update-bfd4f2?style=flat" alt="pr needs update" style="vertical-align: middle"/>

The PR needs to be updated. Either there was a reponse / review by the maintainer or the PR has merge conflicts

<img src="https://img.shields.io/badge/blocked-1d76db?style=flat" alt="blocked" style="vertical-align: middle"/>

This PR is blocked by some other issue. Check regularly if that issue is resolved

<img src="https://img.shields.io/badge/rejected-f9d0c4?style=flat" alt="rejected" style="vertical-align: middle"/>

The PR or the idea of a PR has been rejected by the maintainer. Maybe try again after a few months

<img src="https://img.shields.io/badge/needs%20alternative-F32096?style=flat" alt="needs alternative" style="vertical-align: middle"/>

The package in question needs a replacement that we can bring forward. Feel free to create one. Maitenance burden is on you!

## Where do I start

> - If you like to reach out, look for issues with the <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22needs+first+contact%22"><img src="https://img.shields.io/badge/needs%20first%20contact-b60205?style=flat" alt="needs first contact" style="vertical-align: middle"/></a> label and create an issue (or use some other communication channel). Paste the issue you created as answer to tagged issue here

Reviewer 2: ```suggestion
- If you'd like to reach out, look for issues with the <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22needs+first+contact%22"><img src="https://img.shields.io/badge/needs%20first%20contact-b60205?style=flat" alt="needs first contact" style="vertical-align: middle"/></a> label and create an issue (or use some other communication channel). Paste the issue you created as answer to tagged issue here
```

- If you want to create PRs, look for the <a href="https://github.com/es-tooling/ecosystem-cleanup/issues?q=is%3Aissue+is%3Aopen+label%3A%22accepts+prs%22"><img src="https://img.shields.io/badge/accepts%20prs-DBE1EF?style=flat" alt="accepts prs" style="vertical-align: middle"/></a> label. Paste the PR link in the tagged issue.

> - If you want to own the whole chain, go through:

Reviewer 2: It feels a bit intimidating that there's four steps. I wonder if we could instead provide some general guidance like "For large PRs, you may wish to open an issue in the target repository to ask if the maintainers would be amenable to such a change."

I also think we should mention something like "We generally try to avoid maintainers feeling pressured when creating issues and PRs and thus tackle these issues as individuals rather than mentioning e18e."

Author: Oh i missed that one. The 4 step process is just to showcase the process. I hope its not intimidating since I have the other 2 starting points before that (?).

I actually prefer the mention of e18e. It adds some trust and weight that this kinda is an important topic. I feel like the tone of the issue has to communicate that there is no pressure.
